core/v2/aggregation.go

```diff
@@ -0,0 +1,366 @@
+package corev2
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"slices"
+	"sort"
+
+	"github.com/Layr-Labs/eigenda/chainio"
+	"github.com/Layr-Labs/eigenda/crypto/ecc/bn254"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+const percentMultiplier = 100
+
+const maxNumOperatorAddresses = 300
+
+var (
+	ErrPubKeysNotEqual     = errors.New("public keys are not equal")
+	ErrInsufficientEthSigs = errors.New("insufficient eth signatures")
+	ErrAggPubKeyNotValid   = errors.New("aggregated public key is not valid")
+	ErrAggSigNotValid      = errors.New("aggregated signature is not valid")
+)
+
+type SigningMessage struct {
+	Signature       *bn254.Signature
+	Operator        OperatorID
+	BatchHeaderHash [32]byte
+	// Undefined if this value <= 0.
+	AttestationLatencyMs float64
+	Err                  error
+}
+
+// QuorumResult contains the quorum ID and the amount signed for the quorum
+type QuorumResult struct {
+	QuorumID QuorumID
+	// PercentSigned is percentage of the total stake for the quorum that signed for a particular batch.
+	PercentSigned uint8
+}
+
+// QuorumAttestation contains the results of aggregating signatures from a set of operators by quorums
+// It also returns map of all signers across all quorums
+type QuorumAttestation struct {
+	// QuorumAggPubKeys contains the aggregated public keys for all of the operators each quorum,
+	// including those that did not sign
+	QuorumAggPubKey map[QuorumID]*bn254.G1Point
```

Comment:
plural versus singular: `QuorumAggPubKeys`?

I'm also thinking if there's better ways to arrange these structs (`QuorumResult`, `QuorumAttestation`, and `SignatureAggregation`). It probably doesn't make material differences though